src/model/models/modelManager.ts

```diff
@@ -61,4 +65,45 @@ export class ModelManager {
       throw new AppError('', httpStatus.INTERNAL_SERVER_ERROR, 'store-trigger service is not available', true);
     }
   }
+
+  public async deleteModel(identifier: string): Promise<StoreTriggerResponse> {
+    this.logger.debug({ msg: 'delete record', modelId: identifier });
+    try {
+      const record: Record3D | undefined = await this.catalog.getRecord(identifier);
+      if (record === undefined) {
+        this.logger.error({ msg: 'model identifier not found', modelId: identifier });
+        throw new AppError('NOT_FOUND', httpStatus.NOT_FOUND, `Identifier ${identifier} wasn't found on DB`, true);
+      }
+      if (record.productStatus != RecordStatus.UNPUBLISHED) {
+        this.logger.error({ msg: 'model with status PUBLISHED cannot be deleted', modelId: identifier });
```

Comment:
This is an inner log. There is no need to write it as a client message.
```suggestion
        this.logger.error({ msg: 'got UNPUBLISHED model', modelId: identifier });
```